source/PlayerInfo.cpp

```diff
@@ -2579,3 +2621,46 @@ bool PlayerInfo::CanBeSaved() const
 {
 	return (!isDead && planet && system && !firstName.empty() && !lastName.empty());
 }
+
+
+
+// Called by conversation callbacks to destroy or capture the player's flagship.
+// Either method results in the death of the player.
+void PlayerInfo::Die(int response)
+{
+	if(GetPlanet() || !flagship)
+		Die(true);
+	else
+	{
+		Die();
+		auto it = find(ships.begin(), ships.end(), flagship);
+		if(response == Conversation::EXPLODE)
+			flagship->Destroy();
+		else
+		{
+			flagship->SetIsYours(false);
+			// If the flagship is boarding an NPC, the player's death results in its
+			// capture. Otherwise, consider the player to have died due to a mutiny.
+			// Change the flagship government to remove it from the EscortDisplay HUD.
+			if(boardingShip)
+			{
+				flagship->SetSwizzle(boardingShip->GetSwizzle());
```

Comment:
Ships don't actually change swizzle when they're captured, since that requires a new paint job. (They don't change until you land on a planet.)

Reply:
Indeed, here it was added solely as a visual indicator of "what happened," even if the mission-writer didn't really expound on the details.